### DISCUSSION TITLE
Why any word starting with a `.` drops the top of the stack?

### DISCUSSION THREAD

**Discussion by programandala-net** [arturo-lang/arturo#1288] [Q&A]
I have discovered any word starting with a `.`, even a single `.`, drops (better, gets) the top of the stack:

```
$> 1 2 3 4 5 stack
=> [1 2 3 4 5]

$> .blah
=> 5

$> .
=> 4

$> stack
=> [1 2 3]
```
Tried also in a source file:

```
1 2 3
stack print
. print
.this print
.that print

; Output:

; 1 2 3
; 3
; 2
; 1
```

Is it a side effect of the attribute syntax, an undocumented/unfinished feature or a bug?

**Comment by drkameleon** [comment]
Good catch!

**Let me explain:**

- Basically, `.x` or whatever (that is: **_[attributes](https://arturo-lang.io/documentation/language/#attributes)_** - the same is valid for _attribute labels_ too) are pushed onto a different stack altogether: the attribute stack. So, per se, they don't mess with our main value stack whatsoever

- What you are seeing has nothing to do with how Arturo behaves but with something related to the REPL: basically, the way it's set up, it expects that every single command will yield a value of some kind, so it tries to pop it from the stack and present it as the result of the operation. This is valid in 99% of the cases, but obviously your example is a case where it totally breaks. Btw, here is the relevant part in the implementation: https://github.com/arturo-lang/arturo/blob/master/src/scripts/console.art#L207 (not very elegant, I guess - feel free to suggest anything else! 😉 )

**Reply by drkameleon** [reply]
Regarding the 2nd case, to get my point: just try the exact same thing *without* the attributes and you'll get the exact same result :)
